### PR TITLE
Updated "chrome.downloads"

### PR DESCRIPTION
**Pull request by avi12**
According to https://developer.chrome.com/docs/extensions/reference/downloads

### CODE REVIEW

File: types/chrome/index.d.ts [hunk 1]
```diff
@@ -2627,9 +2627,9 @@ declare namespace chrome.downloads {
         /** Optional. Extra HTTP headers to send with the request if the URL uses the HTTP[s] protocol. Each header is represented as a dictionary containing the keys name and either value or binaryValue, restricted to those allowed by XMLHttpRequest.  */
         headers?: HeaderNameValuePair[] | undefined;
         /** Optional. The HTTP method to use if the URL uses the HTTP[S] protocol.  */
-        method?: string | undefined;
+        method?: "GET" | "POST";
```
Comment (nicolas377) [inline types/chrome/index.d.ts:2630]: you should add `undefined` as an option here

```suggestion
        method?: "GET" | "POST" | undefined;
```

File: types/chrome/index.d.ts [hunk 2]
```diff
@@ -2627,9 +2627,9 @@ declare namespace chrome.downloads {
...
         /** Optional. The action to take if filename already exists.  */
-        conflictAction?: string | undefined;
+        conflictAction?: "uniquify" | "overwrite" | "prompt";
```
Comment (nicolas377) [inline types/chrome/index.d.ts:2632]: same thing over here

```suggestion
        conflictAction?: "uniquify" | "overwrite" | "prompt" | undefined;
```

**Comment by nicolas377** [comment]
Hey @avi12! You should probably modify tests for this change. It makes it a lot easier for us to merge this PR. Also, your top comment should follow this format:

`Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.`

**Review by nicolas377 — CHANGES_REQUESTED** [review]
Even though typescript adds undefined into the possible options for optional params, its good form to add undefined yourself. Try adding that along with changing tests.

**Comment by avi12** [comment]
> You should probably modify tests for this change.

Apologies. I'm honestly not used to making PRs.

> Also, your top comment should follow this format:

Apologies. I made the pull request directly from the online VSCode (accessed via "`.`"), and I do not remember it offering me to add a description.

**Comment by nicolas377** [comment]
That's fine!

File: types/chrome/index.d.ts [hunk 3]
```diff
@@ -6865,12 +6873,12 @@ declare namespace chrome.runtime {
...
-                  page?: string | undefined;
-                  persistent?: boolean | undefined;
-              }
-            | undefined;
+        | {
+            scripts?: string[] | undefined;
+            page?: string | undefined;
+            persistent?: boolean | undefined;
+        }
+        | undefined;
```
Comment (avi12) [inline types/chrome/index.d.ts:6881]: This formatting happened due to pressing `Alt+Shift+F`.

File: types/chrome/index.d.ts [hunk 4]
```diff
@@ -6884,11 +6892,11 @@ declare namespace chrome.runtime {
...
-            | {
-                  service_worker: string;
-                  type?: 'module'; // If the service worker uses ES modules
-              }
-            | undefined;
+        | {
+            service_worker: string;
+            type?: 'module'; // If the service worker uses ES modules
+        }
+        | undefined;
```
Comment (avi12) [inline types/chrome/index.d.ts:6899]: This formatting happened due to pressing `Alt+Shift+F`.

File: types/chrome/index.d.ts [hunk 5]
```diff
@@ -2684,11 +2686,17 @@ declare namespace chrome.downloads {
         previous?: string | undefined;
     }
 
+    export type InterruptReason = "FILE_FAILED" | "FILE_ACCESS_DENIED" | "FILE_NO_SPACE" | "FILE_NAME_TOO_LONG" | "FILE_TOO_LARGE" | "FILE_VIRUS_INFECTED" | "FILE_TRANSIENT_ERROR" | "FILE_BLOCKED" | "FILE_SECURITY_CHECK_FAILED" | "FILE_TOO_SHORT" | "FILE_HASH_MISMATCH" | "FILE_SAME_AS_SOURCE" | "NETWORK_FAILED" | "NETWORK_TIMEOUT" | "NETWORK_DISCONNECTED" | "NETWORK_SERVER_DOWN" | "NETWORK_INVALID_REQUEST" | "SERVER_FAILED" | "SERVER_NO_RANGE" | "SERVER_BAD_CONTENT" | "SERVER_UNAUTHORIZED" | "SERVER_CERT_PROBLEM" | "SERVER_FORBIDDEN" | "SERVER_UNREACHABLE" | "SERVER_CONTENT_LENGTH_MISMATCH" | "SERVER_CROSS_ORIGIN_REDIRECT" | "USER_CANCELED" | "USER_SHUTDOWN" | "CRASH";
+
+    export type State = "in_progress" | "interrupted" | "complete";
```
Comment (nicolas377) [inline types/chrome/index.d.ts:2691]: these names probably need a bit more clarification
```suggestion
    export type DownloadInterruptReason = "FILE_FAILED" | "FILE_ACCESS_DENIED" | "FILE_NO_SPACE" | "FILE_NAME_TOO_LONG" | "FILE_TOO_LARGE" | "FILE_VIRUS_INFECTED" | "FILE_TRANSIENT_ERROR" | "FILE_BLOCKED" | "FILE_SECURITY_CHECK_FAILED" | "FILE_TOO_SHORT" | "FILE_HASH_MISMATCH" | "FILE_SAME_AS_SOURCE" | "NETWORK_FAILED" | "NETWORK_TIMEOUT" | "NETWORK_DISCONNECTED" | "NETWORK_SERVER_DOWN" | "NETWORK_INVALID_REQUEST" | "SERVER_FAILED" | "SERVER_NO_RANGE" | "SERVER_BAD_CONTENT" | "SERVER_UNAUTHORIZED" | "SERVER_CERT_PROBLEM" | "SERVER_FORBIDDEN" | "SERVER_UNREACHABLE" | "SERVER_CONTENT_LENGTH_MISMATCH" | "SERVER_CROSS_ORIGIN_REDIRECT" | "USER_CANCELED" | "USER_SHUTDOWN" | "CRASH";

    export type DownloadState = "in_progress" | "interrupted" | "complete";
```

File: types/chrome/index.d.ts [hunk 6]
```diff
@@ -9444,7 +9452,7 @@ declare namespace chrome.tabs {
  * Permissions:  "tabGroups"
  * @since Chrome 89. Manifest V3 and above.
  */
- declare namespace chrome.tabGroups {
+declare namespace chrome.tabGroups {
```
Comment (nicolas377) [inline types/chrome/index.d.ts:9455]: Undo that change as well
```suggestion
 declare namespace chrome.tabGroups {
```

File: types/chrome/index.d.ts (hunk 4 above)
Comment (nicolas377) [inline types/chrome/index.d.ts:6899]: Undo the formatting change
```suggestion
            | {
                  service_worker: string;
                  type?: 'module'; // If the service worker uses ES modules
              }
            | undefined;```

File: types/chrome/index.d.ts (hunk 3 above)
Comment (nicolas377) [inline types/chrome/index.d.ts:6881]: Undo that change
```suggestion
            | {
                  scripts?: string[] | undefined;
                  page?: string | undefined;
                  persistent?: boolean | undefined;
              }
            | undefined;
```

**Review by nicolas377 — CHANGES_REQUESTED** [review]

File: types/chrome/index.d.ts (hunk 1 above)
Comment (nicolas377) [reply types/chrome/index.d.ts:2632]: Just add `undefined` and you're good!
```suggestion
        method?: "GET" | "POST" | undefined;
```

**Review by nicolas377 — CHANGES_REQUESTED** [review]

**Review by nicolas377 — APPROVED** [review]

**Comment by nicolas377** [comment]
@orta This PR is good to go! Could you approve it?

**Review by userTim — APPROVED** [review]

**Review by orta — APPROVED** [review]
Thanks!

**Comment by avi12** [comment]
Ready to merge

**Pull request merged**